src/main/java/de/idrinth/waraddonclient/service/FileWatcher.java

```diff
@@ -28,10 +29,12 @@ public FileWatcher(GuiAddonList addonList, BaseLogger logger, Config config) thr
             path.mkdirs();
         }
         watcher = path.toPath().getFileSystem().newWatchService();
-        WatchEvent.Kind[] modes = new WatchEvent.Kind[2];
-        modes[0] = StandardWatchEventKinds.ENTRY_CREATE;
-        modes[1] = StandardWatchEventKinds.ENTRY_MODIFY;
-        path.toPath().register(watcher, modes);
+        path.toPath()
+        .register(
+            watcher,
+            Arrays.asList(
+                    StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY)
+                .toArray(new WatchEvent.Kind<?>[] {}));
```

Comment:
this actually looks worse than before, why was this changed?

Reply:
The goal of this change was to first add the type parameter to `WatchEvent`.  I may have been overzealous and attempted to remove `modes` as an unnecessary variable assignment at the same time. I must not have been thinking clearly at the time, as the following is more inline with what I was going for:
```
        watcher = path.toPath().getFileSystem().newWatchService();
        path.toPath().register(watcher, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
```

However, if the preference is to keep it closer to the original structure, I can instead change it to:
```
        WatchEvent.Kind<?>[] modes = new WatchEvent.Kind[2];
        modes[0] = StandardWatchEventKinds.ENTRY_CREATE;
        modes[1] = StandardWatchEventKinds.ENTRY_MODIFY;
```

Reply:
both options are fine for me

Reply:
pushed commit `7e86bf8801661baf7eb8945ecbb0aa0ac763a234` containing the first option.